filesystem/Upload.php

```diff
@@ -131,7 +131,26 @@ public function load($tmpFile, $folderPath = false) {
 			$this->errors[] = _t('File.NOFILESIZE', 'File size is zero bytes.');
 			return false;
 		}
-
+		
+		if (!empty($tmpFile['type'])) {
+            $type = $tmpFile['type'];
+            if (strpos($type, 'image/') !== false) {
+                $tmp_name = $tmpFile['tmp_name'];
+                try {
+                    $test = new ImagickBackend($tmp_name);
+                } catch (Exception $e) {
+					SS_Log::log($e, SS_Log::ERR);
+                    if ($controller = Controller::curr()) {
+                        if ($controller->request->isAjax()){
+                            return $controller->httpError(500, 'Image corrupted');
```

Comment:
load() should not be responsible for prompting a routing redirect. This should be caught at the validation level, and have this bubble up to the user via a field error, rather than redirecting the user. That way the user doesn't have a redirect reset their unsaved form.

Do you want to try moving this to the validator class?